File: 0797-rabbits-in-forest/0797-rabbits-in-forest.py
class Solution:
    def numRabbits(self, answers: List[int]) -> int:
        c = Counter(answers)
        ans = 0
        for x,y in c.items():
            grp_size = x+1
            number_of_groups = (y+x)//grp_size
            ans += number_of_groups*grp_size
        return ans
        

File: 0797-rabbits-in-forest/0797-rabbits-in-forest.java
class Solution {
    public int numRabbits(int[] answers) {
        HashMap <Integer,Integer> map = new HashMap<>();

        for(int i:answers){
            map.put(i,map.getOrDefault(i,0)+1);
        }
        int ans = 0;
        for(int x:map.keySet()){
            int group_size = x+1;
            int number_of_group = (map.get(x)+x) / group_size;
            ans = ans + (number_of_group*group_size);
        }
        return ans;
    }
}